flag.go

```diff
@@ -224,6 +225,16 @@ func (f *FlagSet) Lookup(name string) *Flag {
 	return f.formal[name]
 }
 
+// Mark a flag deprecated in your program
+func (f *FlagSet) MarkDeprecated(name string) error {
```

Comment:
same comment as subcommands - maybe there should be a string to help users figure out why it was deprecated, when it is going away, and what to use instead?